pkg/operator/genericoperatorclient/dynamic_operator_client.go

```diff
@@ -47,6 +47,36 @@ func (c dynamicOperatorClient) Informer() cache.SharedIndexInformer {
 	return c.informer.Informer()
 }
 
+func (c dynamicOperatorClient) GetObjectMeta() (*metav1.ObjectMeta, error) {
+	uncastInstance, err := c.informer.Lister().Get(globalConfigName)
+	if err != nil {
+		return nil, err
+	}
+	instance := uncastInstance.(*unstructured.Unstructured)
+	return getObjectMetaFromUnstructured(instance.UnstructuredContent())
+}
+
+func (c dynamicOperatorClient) UpdateObjectMeta(resourceVersion string, meta *metav1.ObjectMeta) (*metav1.ObjectMeta, error) {
+	uncastOriginal, err := c.informer.Lister().Get(globalConfigName)
+	if err != nil {
+		return nil, err
+	}
+	original := uncastOriginal.(*unstructured.Unstructured)
+
+	copy := original.DeepCopy()
+	copy.SetResourceVersion(resourceVersion)
+	if err := setObjectMetaFromUnstructured(copy.UnstructuredContent(), meta); err != nil {
+		return nil, err
+	}
+
+	ret, err := c.client.Update(context.TODO(), copy, metav1.UpdateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return getObjectMetaFromUnstructured(ret.UnstructuredContent())
+}
+
 func (c dynamicOperatorClient) GetOperatorState() (*operatorv1.OperatorSpec, *operatorv1.OperatorStatus, string, error) {
```

Comment:
As an alternative approach, I could return the metadata from here (instead of only the `resourceVersion`).